flowise-embed/src/components/Bot.tsx

```diff
@@ -368,6 +379,38 @@ export const Bot = (botProps: BotProps & { class?: string }) => {
     handleSubmit(prompt);
   };
 
+  createEffect(() => {
+    const lastSelectionMessage = messages().findLast((message) => message.type === 'selectionMessage')?.message;
+    const lastUserMessage = messages().findLast((message) => message.type === 'userMessage')?.message;
+
+    if (
+      [messageUtils.NCM_INITIAL_QUESTION, messageUtils.NCM_CONTINUE_QUESTION, messageUtils.NCM_HELP_QUESTION].includes(
+        lastSelectionMessage ? lastSelectionMessage : '',
+      )
+    ) {
+      if (lastUserMessage === messageUtils.SIM) {
+        setIsNcmDiscoveringStep(true);
+      } else if (lastUserMessage === messageUtils.NAO) {
+        setIsNcmDiscoveringStep(false);
+      }
```

Comment:
Pode simplificar:
```suggestion
      setIsNcmDiscoveringStep(lastUserMessage === messageUtils.YES);
```
(Como os demais itens no messageUtils estão em inglês, é importante manter o padrão)